### ISSUE TITLE
Moar modifies prompt characters

### ISSUE THREAD

**Issue by kalocsaibotond** [walles/moar#258]
I tried out moar in (void) linux console with `moar hello.sh` and for some reason it modifies the characters of the prompt:
![moar_console_bug PNG](https://github.com/user-attachments/assets/a2e8f5e3-06bb-44df-98ff-8ca6c292fd6c)
Interestingly `eza` or `ls`resets the prompt characters.

For curiosity, I tried out bat with moar as pager I also get a similar but different prompt character modification:
![moar_bat_console_bug PNG](https://github.com/user-attachments/assets/bd707162-751b-4f0d-95c0-89b55ca277d6)
In console `eza` or `ls` resets the characters.
For bat `export BAT_PAGER='moar -no-linenumbers -quit-if-one-screen` .

The latter moar bat behaviour is also present in simple terminal:
![moar_bat_st_bug PNG](https://github.com/user-attachments/assets/af255f58-71c9-4f55-9149-12d777a42537)
However only `eza` resets the character.

For the same use cases, `less` do not present these behaviours.

I do not experience these behaviours in windows 10 and 11 in the latest WezTerm.

**Comment by walles** [comment]
1. Can you do `moar --trace hello.sh`, reproduce the problem and post the trace output here?
2. What terminal are you using when you reproduce this?
3. How do you exit `moar`? By pressing `q`? Pressing `ESC`? Something else?

**Comment by kalocsaibotond** [comment]
> 1. Can you do `moar --trace hello.sh`, reproduce the problem and post the trace output here?

The result of `moar --trace hello.sh` :
![moar_console_bug_trace](https://github.com/user-attachments/assets/8e5a5ab2-66a1-4bad-8e80-464ed4b050d9)

> 2. What terminal are you using when you reproduce this?

I made the first and second image in my description in the default linux console of Void Linux (base 2024-03-14 ) . I made third image was in the suckless Simple Terminal . I also made the above image in the default linux console .

> 3. How do you exit `moar`? By pressing `q`? Pressing `ESC`? Something else?

I used the 'q' button. Though trying the 'ESC' also yields the same result.

**Comment by walles** [comment]
1. Are you running something like tmux, screen or [ct](https://github.com/hSaria/ChromaTerm) that could affect moar's behavior?
2. Do you have any terminal settings that could affect moar's behavior? 

In your image (see below), moar exits, but moar's UI is still there right before the prompt.

This is weird, since moar uses the [Alternative Screen](https://ratatui.rs/concepts/backends/alternate-screen/) for all its drawing. When moar exits it switches back.

Switching back should at least make moar's UI disappear before moar terminates.

If the alternate screen implementation in the terminals where you reproduced this was broken, that could explain it, but:
* You tried with two different terminals. Both of them failing to implement altscreen seems unlikely.
* I tried with the same terminals, and the problems didn't appear for me

![moar_console_bug PNG](https://github.com/user-attachments/assets/a2e8f5e3-06bb-44df-98ff-8ca6c292fd6c)

**Comment by kalocsaibotond** [comment]
> 1. Are you running something like tmux, screen or [ct](https://github.com/hSaria/ChromaTerm) that could affect moar's behavior?
> 2. Do you have any terminal settings that could affect moar's behavior?
> 
> In your image (see below), moar exits, but moar's UI is still there right before the prompt.
> 
> This is weird, since moar uses the [Alternative Screen](https://ratatui.rs/concepts/backends/alternate-screen/) for all its drawing. When moar exits it switches back.
> 
> Switching back should at least make moar's UI disappear before moar terminates.
> 
> If the alternate screen implementation in the terminals where you reproduced this was broken, that could explain it, but:
> 
> * You tried with two different terminals. Both of them failing to implement altscreen seems unlikely.
> * I tried with the same terminals, and the problems didn't appear for me
> 
> ![moar_console_bug PNG](https://private-user-images.githubusercontent.com/90713156/391923675-a2e8f5e3-06bb-44df-98ff-8ca6c292fd6c.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2NjQyNzgsIm5iZiI6MTczMzY2Mzk3OCwicGF0aCI6Ii85MDcxMzE1Ni8zOTE5MjM2NzUtYTJlOGY1ZTMtMDZiYi00NGRmLTk4ZmYtOGNhNmMyOTJmZDZjLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDEzMTkzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5YWFkMjIwYTZiMmJlZjVmM2U5MmZmODNiZDY5MmI5YjU2M2M0NWJiNTY5NzQzYzgyMjI5NGJkZmI1MjNhMWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l6O4LgaO7BUnfzvK8_7iZeVeCDgvpZxUN9iMJ7VBSTs)

No, my setup for the first bug is the simplest possible: I downloaded the [void live `image void-live-x86_64-20240314-base.iso`](https://repo-default.voidlinux.org/live/current/void-live-x86_64-20240314-base.iso). I started it on a Qemu virtual machine. I synchronised the xbps repository index. I installed `moar` and used it on a file. I did not use anything apart from the vanilla linux console (no tmux, no screen, no ct, no xorg, no wayland, no etc..)

Now the second bug requires [`bat`](https://github.com/sharkdp/bat) and the presence of
```sh
export BAT_PAGER='moar -no-linenumbers -quit-if-one-screen'
```
environment variable. It is also reproducible with the pure base void live image if `moar`, `bat` is installed and `BAT_PAGER` environment variable is set. This can be induced by viewing a file with `bat` like `bat hello.sh` .

However, unlike the first bug, it is reproducible outside the linux console. So far I reproduced it in st and WezTerm. Here is the reproduction of the bug with bat on Windows 11 (Powershell 7.4.6), in WezTerm (wezterm 20240203-110809-5046fc22):
![moar_bat_wezterm_bug PNG](https://github.com/user-attachments/assets/482c8c6f-8e8a-4379-bbcb-bb853846ef7e)
